dns/provider.go

```diff
@@ -90,6 +90,31 @@ func Provider() terraform.ResourceProvider {
 							Optional:    true,
 							DefaultFunc: schema.EnvDefaultFunc("DNS_UPDATE_KEYSECRET", nil),
 						},
+						"gssapi": &schema.Schema{
+							Type:        schema.TypeBool,
+							Optional:    true,
+							DefaultFunc: schema.EnvDefaultFunc("DNS_UPDATE_GSSAPI", nil),
```

Comment:
If the GSSAPI arguments are mutually-exclusive with the existing `key_` settings, we can tell the SDK that with an additional field here:

```go
    ConflictsWith: []string{"update.key_name", "update.key_algorithm", "update.key_secret"},
```

Specifying that will ask the SDK to validate that these are mutually exclusive and return an error to the user automatically if not.

Reply:
That should make things much simpler.